lib/ansible/modules/database/postgresql/pg_hba.py

```diff
@@ -0,0 +1,769 @@
+        elif ipv6_re.search(rule['src']):
+            if '/' in rule['src']:
+                # prefix tells how much 1's there are in netmask, so lets use that for sourceweight
+                prefix = rule['src'].split('/')[1]
+                srcweight = int(prefix)
+            elif 'mask' in rule.keys():
+                # Netmask. Let's count the 1's in the netmask in binary form.
+                bits = "{0:b}".format(ipv6_to_int(rule['mask']))
+                srcweight = bits.count('1') * 4
+            else:
+                # seems, there is no netmask / prefix to be found. Then only one IP applies.
+                srcweight = 128  # (ipv4 /32 is considered equivalent to ipv6 /128)
+        else:
+            # You can also write all to match any IP address, samehost to match any of the server's own IP addresses,
+            # or samenet to match any address in any subnet that the server is directly connected to.
+            if rule['src'] == 'all':
+                srcweight = 0
+            elif rule['src'] == 'samehost':
+                srcweight = 128  # (ipv4 /32 is considered equivalent to ipv6 /128)
+            elif rule['src'] == 'samenet':
+                # Might write some fancy code to determine all prefix's
+                # from all interfaces and find a sane value for this one.
+                # For now, let's assume /24...
+                srcweight = 96  # (ipv4 /24 is considered equivalent to ipv6 /96)
+            elif rule['src'][0] == '.':
+                # suffix matching, let's asume a very large scale and therefore a very low weight.
+                srcweight = 64  # (ipv4 /16 is considered equivalent to ipv6 /64)
+            else:
+                # hostname, let's asume only one host matches
+                srcweight = 128  # (ipv4 /32 is considered equivalent to ipv6 /128)
+
+        # One little thing: for db and user weight, higher weight means less specific and thus lower in the file.
+        # Since prefix is higher for more specific, I inverse the output to align with how dbweight and userweight works...
+        srcweight = 128 - srcweight  # (higher prefix should be lower weight)
+
+        if rule['db'] == 'all':
+            dbweight = len(self.databases) + 1
+        elif rule['db'] == 'replication':
+            dbweight = 0
+        elif rule['db'] in ['samerole', 'samegroup']:
+            dbweight = 1
+        else:
+            dbweight = 1 + rule['db'].count(',')
+
+        if rule['usr'] == 'all':
+            uweight = len(self.users) + 1
+        else:
+            uweight = 1
+
+        ret = []
+        for c in self.order:
+            if c == 'u':
+                ret.append(uweight)
+            elif c == 's':
+                ret.append(srcweight)
+            elif c == 'd':
+                ret.append(dbweight)
+
+        return tuple(ret)
+
+    def is_empty(self):
+        if len(self.rules) == 0:
```

Comment:
The if/else block be replaced by `return not self.rules`.

Reply:
I have enabled pylint and he tells me the same. Will fix all errors. This is fixed.